src/DataGrid.tsx

```diff
@@ -490,7 +474,9 @@ function DataGrid<R, SR, K extends Key>(
     onExpandedGroupIdsChange(newExpandedGroupIds);
   }
 
-  function onGridFocus() {
+  function onGridFocus({ target }: React.FocusEvent<HTMLDivElement>) {
+    // Do not scroll if clicked within the grid
+    if (gridRef.current !== target && gridRef.current!.contains(target)) return;
```

Comment:
I am not sure if this can be an issue. One workaround is to always render the selected column and row and remove `onGridFocus` handler